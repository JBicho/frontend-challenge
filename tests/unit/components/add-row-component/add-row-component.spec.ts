import Vue from 'vue';
import { mount, shallowMount, createLocalVue } from '@vue/test-utils';
import AddRow from '@/components/add-row-component/add-row-component.vue';
import Vuex from 'vuex';

const localVue = createLocalVue();
localVue.use(Vuex);
Vue.use(Vuex);

describe('Add Row Component Tests', () => {
  let wrapper: any;
  let store: any;
  let actions: any;
  let state: any;

  beforeEach(() => {
    actions = {
      addTranscription: jest.fn(() => true)
    };

    state = {
      transcriptionList: [],
      errors: []
    };

    store = new Vuex.Store({
      actions,
      modules: {
        transcription: {
          namespaced: true,
          state,
          actions
        }
      }
    });
  });

  test('Snapshot test', () => {
    const Constructor = Vue.extend(AddRow);
    const vm = new Constructor().$mount();
    expect(vm.$el).toMatchSnapshot();
  });

  test('Has a button', () => {
    wrapper = mount(AddRow);

    expect(wrapper.contains('.add-row-button')).toBe(true);
  });

  test('Passes action to store', () => {
    const instance = shallowMount(AddRow, { store, localVue });

    instance.find('.add-row-button').trigger('click');

    expect(actions.addTranscription).toHaveBeenCalled();
  });
});
