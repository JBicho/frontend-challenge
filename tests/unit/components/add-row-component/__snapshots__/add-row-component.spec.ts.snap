// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Row Component Tests Snapshot test 1`] = `
<div
  class="add-row"
>
  <button
    class="add-row-button"
  >
    <svgicon
      height="1.7rem"
      name="add-row"
      original="true"
      width="1.7rem"
    />
  </button>
</div>
`;
